wrappers/ios/demo/demo/ViewController.swift

```diff
@@ -35,11 +40,13 @@ class ViewController: UIViewController {
             self.captureSession.addInput(cameraInput)
             let videoDataOutput = AVCaptureVideoDataOutput()
             videoDataOutput.setSampleBufferDelegate(self, queue: self.queue)
-            videoDataOutput.videoSettings = [kCVPixelBufferPixelFormatTypeKey as String: Int(kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange)]
+            videoDataOutput.videoSettings = [kCVPixelBufferPixelFormatTypeKey as String: kCVPixelFormatType_32BGRA]
```

Comment:
I have a technical counter argument for this change: first of all, looking at the setup of the `ImageView` where only `ImageFormat::Lum` is used, I would assume this is simply a bug. Second of all, it is simply a waste of CPU resources to transform RGB data to grey scale inside zxing-cpp when we can have the latter right from the start.

Reply:
I was basically referring to that method here https://github.com/zxing-cpp/zxing-cpp/blob/master/wrappers/ios/Sources/Wrapper/Reader/ZXIBarcodeReader.mm#L56 and thought it would be fine using BGRA.

The reason to change that is that modern devices natively output 32BGRA. So using YUV is an additional conversion on the phone and - if my knowledge is not completely wrong - is lossy. But maybe that doesn't do any harm, I can easily revert it back.

Reply:
I see, so the 'bug' part of my comment was b.s. But the alternative path then explicitly asks the OS to convert the input into an `kCGColorSpaceGenericGray` image, which is then fed into `ReadBarcodes`. It would be possible to pass `BGRA` pixels directly but they would then only be converted to grey scale data internally. And having an OS level function do that is much more likely to be optimal in terms of CPU usage.

All that said, I could imagine a situation where converting `BGRA` to greyscale is in-fact cheaper than asking for `kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange` if the original raw data was in fact already `BGRA` because the two chroma channels are then computed for nothing. But if you are running on some hardware where the 'raw' data you get out of your sensor/preprocessor is Yuv, then simply passing the Y-channel is obviously the most effective option. And I understood that this is actually the most prevalent situation in the mobile phone space?

Reply:
I changed it back, there was no need to do the `Int()` cast.

Reply:
@axxel Still not quite sure how your internal binarizer works with RGB images and if that makes really a difference but I was kind of right:

https://developer.apple.com/documentation/corevideo/kcvpixelformattype_420ypcbcr8biplanarvideorange

Lum in the YUV format only ranges from 16 to 235.

Reply:
As far as I remember, @ChristianNorbertBraun and I checked the native format of some iPhones when developing the code, and they supported `kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange` as native format. From our understanding, this is the most efficient and broadly supported format. It may have changed for newer iPhones though, but I assume that it still true.